website/blog/2024-08-26-how-we-run-python-using-webassembly/index.mdx

```diff
@@ -0,0 +1,91 @@
+# Running Python code with WebAssembly
+
+Language runtimes are nice, but integrating one is an entire story. Let's discover how we managed to integrate the Python runtime within the Metatype ecosystem.
+
+## Why?
+
+You have probably heard of "Function as a Service" or FaaS. In simple terms, FaaS enables anyone to run code in response to events without the hassle of managing the underlying infrastructure needed for microservices.
+
+This capability was initially added into Metatype through Deno workers, which provided ways to execute functions authored in Typescript with ease.
+
+On the other hand, people might also want to write functions in Python, which, as you may know, is very popular, and its standard library can be quite resourceful for this type of use case.
+
+However, integrating a Python runtime isn't as straightforward as integrating Deno.
+
+## What are the constraints?
+
+There are few constraints that had to be met:
+
+1. **Security**: functions should have limited access to the execution evironment
+2. **Speed**: functions should run with low latency
+3. **User-friendliness**: writing a function in one language should be as near identical as writing one in another
+
+## Choosing an approach...
+
+Let's start simple, we want to run Python code, so what do we need?
+
+- A Python runtime
+
+We do not want to be platform-dependent, but how can we achieve that?
+
+- Docker
+- WebAssembly
+
+At a higher level, Docker and WebAssembly all share the same idea: package an application into one build and make it platform independent.
+
+## The WebAssembly Interface (WASI)
+
+This is basically WebAssembly but with more capabilities such as file system access and networking.
+
+Just like x86, or Wasm, WASI is a compilation target.
+The **build** can then be executed in a sandboxed environment within a WASI-compliant runtime (WasmEdge, Wasmtime, Wasmer).
+
+This fits very well to our use case, and as opposed to Docker, we can instantiate, manage and expose ressources programmatically with ease.
+
+And luckily for us at the time, some [community works](https://github.com/vmware-labs/webassembly-language-runtimes) has already been done on enabling the use of Python into WASI.
```

Comment:
I don't think "work" should be plural here.

Reply:
You're right, it's surprising that CodeRabbit didn't catch that!